### PR TITLE
[DPE-3184] Update secrets implementation

### PR DESCRIPTION
**Pull request by dragomirp**
Update peer secrets implementation to be up to date with other PG charms

### CODE REVIEW

File: pyproject.toml
Comment (dragomirp) [inline pyproject.toml:1]: Missed the linter configs when ruff was updated

File: src/relations/peers.py [hunk 1]
```diff
@@ -237,17 +237,3 @@ def update_auth_file(self, auth_file: str) -> None:
...
-            return
-
-        self.charm.set_secret(APP_SCOPE, username, password)
-
-    def remove_user(self, username: str):
-        """Removes user from app databag."""
-        if not self.charm.unit.is_leader():
-            return
-
-        self.charm.set_secret(APP_SCOPE, username, None)
```
Comment (dragomirp) [inline src/relations/peers.py:253]: Removing arbitrary secret fields as with the k8s charm

File: src/charm.py [hunk 2]
```diff
@@ -233,183 +255,59 @@ def _normalize_secret_key(self, key: str) -> str:
...
-            return self._juju_secret_remove(scope, key)
-        if scope == UNIT_SCOPE:
-            del self.peers.unit_databag[key]
+        peers = self.model.get_relation(PEER_RELATION_NAME)
+        secret_key = self._translate_field_to_secret_key(key)
+        if scope == APP_SCOPE:
+            self.peer_relation_app.delete_relation_data(peers.id, [secret_key])
         else:
-            del self.peers.app_databag[key]
+            self.peer_relation_unit.delete_relation_data(peers.id, [secret_key])
```
Comment (juditnovak) [inline src/charm.py:310]: I really should have introduced a 
```
def peer_relation_data(scope: Scope) -> DataPeer:
    if scope == APP_SCOPE:
        return self.peer_relation_app
    elif scope == APP_UNIT:
        return self.peer_relation_unit     
```
function, to turn the above (repetitive) structure to
```
    self.peer_relation_data(scope).delete_relation_data(peers.id, [secret_key])
```

If you've got the capacity, don't U wanna do that? If not, of course feel free to ignore (it's cosmetics)

**Review by juditnovak — APPROVED** [review]

**Review by juditnovak — APPROVED** [review]
Even more approved than one round earlier :-)

**Pull request merged**